Point-of-No-Return/Collision.cpp

```diff
@@ -0,0 +1,108 @@
+﻿#include "Collision.h"
+
+#include <vector>
+
+#include "Object.h"
+
+std::vector<Vec2> Collision::SearchBlock(const Vec2& character_pos,const Size& size,int** map)
+{
+	Vec2 vec2(0,0);
+	std::vector<Vec2> mapdata;
+
+	int center_x = character_pos.x.value + size.width.value / 2;
+	int center_y = character_pos.y.value + size.height.value / 2;
+
+	Matrix search_start = 
+	{
+		static_cast<Col>(center_x / MapChipInfo::chip_size - 2),
+		static_cast<Row>(center_y / MapChipInfo::chip_size - 2)
+	};
+	
+	for (int i = 0; i < 5;i++)
+	{
+		for (int j = 0;j < 5;j++)
+		{
+			int current_col = search_start.col.value + j;
+			int current_row = search_start.row.value + i;
+
+			if (map[current_row][current_col] != 0)
+			{
+				vec2.x.value = current_col * MapChipInfo::chip_size;
+				vec2.y.value = current_row * MapChipInfo::chip_size;
+				mapdata.push_back(vec2);
+			}
+		}
+	}
+	return mapdata;
+}
+
+//
+void Collision::CheckBlock(Character& character, Vec2 characterPrevious, Size characterSize, Vec2 characterPosition, std::vector<Vec2> blockPositions)
+{
+	Vec2 vector =
+	{
+		characterPrevious.x.value - characterPosition.x.value,
+		characterPrevious.y.value - characterPosition.y.value
+	};
+
+	// 修正する方向を入れる変数
+
+	for (auto blockPosition: blockPositions)
+	{
+		float characterLeft = characterPosition.x.value;
+		float characterRight = characterPosition.x.value + characterSize.width.value;
```

Comment:
```suggestion
		float characterRight = characterLeft + characterSize.width.value;
```
にすべきですね。こういうことしてるとバグの素(もと)です。

演算子オーバーロードのときにも指摘しましたが、DRYに含まれる話だと思います。

Reply:
修正しました。https://github.com/human-osaka-game-2019/Point-of-No-Return/pull/39/commits/8bc6f4515cf68f72672171a397d00def58bdc561